src/omv/modules/py_tf.c

```diff
@@ -829,6 +842,346 @@ STATIC mp_obj_t py_tf_regression(mp_obj_t model_obj, mp_obj_t array_obj) {
+    }
+
+    memcpy(&self_in->bounding_boxes, &nms_bounding_boxes, sizeof(list_t));
+
+    // Create a list per class label.
+    mp_obj_t list = mp_obj_new_list(max_label + 1, NULL);
+    for (size_t i = 0; i <= max_label; i++) {
+        ((mp_obj_list_t *) list)->items[i] = mp_obj_new_list(0, NULL);
+    }
+
+    float x_scale = self_in->roi->w / ((float) self_in->params->input_width);
+    float y_scale = self_in->roi->h / ((float) self_in->params->input_height);
+    // MAX == KeepAspectRatioByExpanding - MIN == KeepAspectRatio
+    float scale = IM_MIN(x_scale, y_scale);
+    int x_offset = fast_floorf((self_in->roi->w - (self_in->params->input_width * scale)) / 2.0f) + self_in->roi->x;
+    int y_offset = fast_floorf((self_in->roi->h - (self_in->params->input_height * scale)) / 2.0f) + self_in->roi->y;
+
+    size_t len = list_size(&nms_bounding_boxes);
+    for (size_t i = 0; i < len; i++) {
+        py_tf_model_output_bounding_box_lnk_data_t lnk_data;
+        list_pop_front(&nms_bounding_boxes, &lnk_data);
+        py_tf_classification_obj_t *o = m_new_obj(py_tf_classification_obj_t);
+        o->base.type = &py_tf_classification_type;
+        o->x = mp_obj_new_int(fast_floorf(lnk_data.rect.x * scale) + x_offset);
+        o->y = mp_obj_new_int(fast_floorf(lnk_data.rect.y * scale) + y_offset);
+        o->w = mp_obj_new_int(fast_floorf(lnk_data.rect.w * scale));
+        o->h = mp_obj_new_int(fast_floorf(lnk_data.rect.h * scale));
+        o->output = mp_obj_new_float(lnk_data.score);
+        mp_obj_list_append(((mp_obj_list_t *) list)->items[lnk_data.label_index], o);
+    }
+
+    return list;
+}
+STATIC MP_DEFINE_CONST_FUN_OBJ_KW(py_tf_model_output_get_bounding_boxes_obj, 1, py_tf_model_output_get_bounding_boxes);
+
+STATIC const mp_rom_map_elem_t py_tf_model_output_locals_table[] = {
+    { MP_ROM_QSTR(MP_QSTR_rect),                MP_ROM_PTR(&py_tf_model_output_rect_obj) },
+    { MP_ROM_QSTR(MP_QSTR_x),                   MP_ROM_PTR(&py_tf_model_output_x_obj) },
+    { MP_ROM_QSTR(MP_QSTR_y),                   MP_ROM_PTR(&py_tf_model_output_y_obj) },
+    { MP_ROM_QSTR(MP_QSTR_w),                   MP_ROM_PTR(&py_tf_model_output_w_obj) },
+    { MP_ROM_QSTR(MP_QSTR_h),                   MP_ROM_PTR(&py_tf_model_output_h_obj) },
+    { MP_ROM_QSTR(MP_QSTR_input_height),        MP_ROM_PTR(&py_tf_model_output_input_height_obj) },
+    { MP_ROM_QSTR(MP_QSTR_input_width),         MP_ROM_PTR(&py_tf_model_output_input_width_obj) },
+    { MP_ROM_QSTR(MP_QSTR_input_channels),      MP_ROM_PTR(&py_tf_model_output_input_channels_obj) },
+    { MP_ROM_QSTR(MP_QSTR_output_height),       MP_ROM_PTR(&py_tf_model_output_output_height_obj) },
+    { MP_ROM_QSTR(MP_QSTR_output_width),        MP_ROM_PTR(&py_tf_model_output_output_width_obj) },
+    { MP_ROM_QSTR(MP_QSTR_output_channels),     MP_ROM_PTR(&py_tf_model_output_output_channels_obj) },
+    { MP_ROM_QSTR(MP_QSTR_value),               MP_ROM_PTR(&py_tf_model_output_value_obj) },
```

Comment:
Can we replace those with attributes ? See example in
`ndarray_properties_attr` in ulab or `ssl_context_attr` in `extmod/modtls_mbedtls.c`
And replace `value()` with `subscr` (for this there are many examples in `modules/py_image.c`)

Reply:
Does subscr support 3D access? I guess I could make the property names shorter, and then it doesn't matter.

Reply:
Done.

Reply:
Note, I removed 3D subscript access. This was slowing things down by 100x.

Reply:
Can we index using a tuple like numpy does ? Not this will be important for multi-output models, example:
```Python
In [28]: import numpy as np
In [29]: a = np.array([[1, 2, 3], [4, 5, 6]])
In [30]: a[(0, 2)]
Out[30]: 3
```

Reply:
No, you can't use any tuples. Their creation causes a malloc. The speed slow-down is ridiculous.

I got to the current way the API works after starting with thinking we could do tuples and realizing it was a hard no.

Reply:
Understood.